src/main/java/org/opensearch/knn/index/KNNVectorScriptDocValues.java

```diff
@@ -45,8 +46,9 @@ public void setNextDocId(int docId) throws IOException {
         try {
             BytesRef value = binaryDocValues.binaryValue();
             ByteArrayInputStream byteStream = new ByteArrayInputStream(value.bytes, value.offset, value.length);
-            ObjectInputStream objectStream = new ObjectInputStream(byteStream);
-            return (float[]) objectStream.readObject();
+            final VectorSerializer vectorSerializer = VectorSerializerFactory.getSerializerByStreamContent(byteStream);
+            final float[] vector = vectorSerializer.byteToFloat(byteStream);
+            return vector;
```

Comment:
we can probably compress the two statements to - 

`return vectorSerializer.byteToFloat(byteStream);`

Reply:
I actually left it on purpose, it makes code more readable and makes it easy to debug/adding log if needed. Compiler will optimize both syntax to the same byte-code anyway